src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -326,6 +326,19 @@ private List<LibraryDependency> GetDependenciesFromSpecRestoreMetadata(PackageSp
                 var targetFrameworkInfo = packageSpec.GetTargetFramework(targetFramework);
                 dependencies.AddRange(targetFrameworkInfo.Dependencies);
 
+                if (packageSpec.RestoreMetadata?.CentralPackageVersionsEnabled ?? false)
+                {
+                    var dependencyNamesSet = new HashSet<string>(targetFrameworkInfo.Dependencies.Select(d => d.Name), StringComparer.OrdinalIgnoreCase);
+                    dependencies.AddRange(targetFrameworkInfo.CentralPackageVersions
+                        .Where(item => !dependencyNamesSet.Contains(item.Key))
+                        .Select(item => new LibraryDependency()
+                        {
+                            LibraryRange = new LibraryRange(item.Value.Name, item.Value.VersionRange, LibraryDependencyTarget.Package),
+                            VersionCentrallyManaged = true,
+                            AutoReferenced = true,
```

Comment:
Added a different concept.

Reply:
I missed to answer to this: "Why in particular do we need this?" 

The provider will be aware of the central package versions but these dependencies needs to not be like the direct ones, they will be "weaker". Their job is to be there in case that a transitive version needs to be updated by a central version and also to catch and downgrades. When during the graph creations the provider is asked to provide the dependencies (https://github.com/NuGet/NuGet.Client/blob/71e16df95a76ce4ff624ecb7c912a479c33c1da9/src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs#L145) the "weaker" dependencies will not be added to the graph. They will be added only as needed (== if a transitive dependency is present in the central list).